POLMED/src/components/nav/index.tsx

```diff
@@ -1,40 +1,53 @@
+import { useEffect, useState } from "react";
 import { Link, useLocation } from "react-router-dom";
-import NavItem from "./navItems";
-import { navItems, NavItemType } from "./navItems/navItems";
 import logo from "@/img/logo/logo.png";
 import whiteLogo from "@/img/logo/logo_white.png";
+import NavUl from "./components/navUl";
 
 const Nav = () => {
   const location = useLocation();
   const { pathname: path } = location;
 
+  const [windowSize, setWindowSize] = useState([window.innerWidth, window.innerHeight]);
+
+  useEffect(() => {
+    const handleWindowResize = () => {
+      setWindowSize([window.innerWidth, window.innerHeight]);
+      if (window.innerWidth > 1024) {
+        const navUl = document.querySelector(".nav-ul");
+        (navUl as HTMLElement).style.setProperty("transform", "translateX(0)");
+      }
+    };
+
+    window.addEventListener("resize", handleWindowResize);
+
+    return () => {
+      window.removeEventListener("resize", handleWindowResize);
+    };
+  });
+
+  const toggleMenu = () => {
+    const navUl = document.querySelector(".nav-ul");
+    console.log(navUl);
+    (navUl as HTMLElement).style.setProperty("transform", "translateX(-100%)");
+  };
```

Comment:
Nie będzie wydajniej i czyściej jak zrobisz to w scss? 
Można by wtedy utworzyć osobny plik z wszystkimi _media queries_.

Reply:
generalnie dzięki temu, mogę zastosować poniżej warunkowe renderowanie poszczególnych rzeczy takich jak ikonka bars, nawigacja mobilna/desktopowa, a następnie w zależności od tego czy windowSize.innerWidth jest < 1024 renderuje ikonki w nawigacji mobilnej i przycisk do wylogowywania się zmienia, dlatego osobiście wolałbym to zostawić.
Może i się jakoś da to obejść bez tego useEffecta, ale szczerze mówiąc myślałem teraz chwile jak inaczej to zrobić i co chwile napotykałem kolejny problem :/